wdl-lint/src/rules/trailing_comma.rs

```diff
@@ -0,0 +1,169 @@
+//! A lint rule for trailing commas in lists/objects.
+
+use wdl_ast::v1::MetadataArray;
+use wdl_ast::AstNode;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::ToSpan;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the trailing comma rule.
+const ID: &str = "TrailingComma";
+
+/// Diagnostic message for missing trailing comma.
+fn missing_trailing_comma(span: Span) -> Diagnostic {
+    Diagnostic::note("item missing trailing comma")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("add a comma after this element")
+}
+
+/// Diagnostic message for extraneous content before trailing comma.
+fn extraneous_content(span: Span) -> Diagnostic {
+    Diagnostic::error("extraneous content before trailing comma")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("remove this extraneous content")
+}
+
+/// Detects missing trailing commas.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct TrailingCommaRule;
+
+impl Rule for TrailingCommaRule {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Ensures that lists and objects have a trailing comma."
+    }
+
+    fn explanation(&self) -> &'static str {
```

Comment:
1. I don't think this should mention the `key_value_pairs` rule (especially not with that casing). This is another case for that "linked rule" idea I proposed somewhere else, but IMO doesn't belong in the `explanation()` text.
2. There's another case not mentioned in this text or your code, but IMO should be covered here. Any `CallStatement` with more than one input should have trailing commas (and also be kept to one input-per-line).

Reply:
I removed #1. For #2, I think that belongs in the `CallInputSpacing` rule, personally.

Reply:
Ehh, maybe not for (2), since that will likely need the `find_next_comma` function.

Reply:
It's the exact same logic applied here, isn't it? More-or-less? That suggests to me the enforcement falls here as well.

A rule with `Spacing` in the name shouldn't enforce comma presence IMO.

Reply:
@a-frantz - OK, I just added the CallStatement check. The spacing is enforced in `CallInputSpacing`. The trailing comma is enforced here. It actually checks each `input` for a trailing comma, even though missing commas before the last element would be a syntax error.